File: src/helpers/retries.ts
export async function withRetry<T>(operation: () => Promise<T>, attempts = 2): Promise<T> {
  let lastError: unknown;

  for (let index = 0; index < attempts; index += 1) {
    try {
      return await operation();
    } catch (error) {
      lastError = error;
      if (index < attempts - 1) {
        await new Promise((resolve) => setTimeout(resolve, 150));
      }
    }
  }

  throw lastError;
}

File: browser_platform/helpers/retries.py
from __future__ import annotations
import asyncio
from typing import Any, Callable, TypeVar

T = TypeVar('T')


async def with_retry(operation: Callable[[], Any], attempts: int = 2) -> Any:
    last_error: BaseException | None = None
    for index in range(attempts):
        try:
            return await operation()
        except Exception as error:
            last_error = error
            if index < attempts - 1:
                await asyncio.sleep(0.15)
    raise last_error
